src/edutap/wallet_google/api.py

```diff
@@ -374,15 +374,21 @@ def save_link(
         for obj in objs:
             # first look if this is an object reference as dict
             if isinstance(obj, dict) and "id" in obj and len(obj.keys()) <= 2:
-                obj = GoogleWalletObjectReference.model_validate(obj)
-            if isinstance(obj, GoogleWalletObjectReference):
-                payload[name].append(obj.model_dump(exclude_none=True, mode="json"))
+                obj = GoogleWalletObjectWithClassReferenceMixin.model_validate(obj)
+            if isinstance(obj, GoogleWalletObjectWithClassReferenceMixin):
+                payload[name].append(
+                    obj.model_dump_json(
+                        exclude_none=True,
+                        exclude_unset=True,
+                        exclude_defaults=True,
+                    )
+                )
                 continue
 
             # otherwise it must be a registered model
             model = lookup_model_by_plural_name(name)
             obj = _validate_data(model, obj)
-            obj_json = obj.model_dump(exclude_none=True, mode="json")
+            obj_json = obj.model_dump_json(exclude_none=True)
```

Comment:
Here I am pretty sure `model_dump_json` works internally different than `model_dump` with `mode=json`. And I changed it to model_dump` because of some serious problems. I just can not remember the details now.